### ISSUE TITLE
[reps] local and session storage missing keys

### ISSUE THREAD

**Issue by MikeRatcliffe** [devtools-html/devtools-core#1032]
Same issue with sessionStorage.

STR:

```
localStorage.setItem("key", "value1")
localStorage.setItem("key2", "value2")
localStorage
```

Look at it in the web console and you will see that only "key2" is displayed.

This is because Object.keys(localStorage) and Object.getOwnPropertyNames(localStorage) both return ["key2"] (not sure which we use in reps).

Why is "key" missing? Because there is a key method on the prototype and we don't want to overwrite it (you can access key2 using `localstorage.key2`)... `localStorage.setItem("setItem", "value3")` would produce a similar issue.

In a nutshell, to get the property names from localStorage and sessionStorage we need  to do this:

```
const props = [];
for (let i = 0; i < localStorage.length; i++) {
  const key = localStorage.key(i);
  props.push(key);
}
```

**Comment by nchevobbe** [comment]
Here is the packet we get for this evaluation result:
```json
{
    "type": "object",
    "actor": "server1.conn1.child1/obj26",
    "class": "Storage",
    "extensible": true,
    "frozen": false,
    "sealed": false,
    "ownPropertyLength": 2,
    "preview": {
      "kind": "Object",
      "ownProperties": {
        "key3": {
          "configurable": true,
          "enumerable": true,
          "writable": true,
          "value": "value2"
        },
        "key2": {
          "configurable": true,
          "enumerable": true,
          "writable": true,
          "value": "value2"
        }
      },
      "ownSymbols": [],
      "ownPropertiesLength": 2,
      "ownSymbolsLength": 0,
      "safeGetterValues": {
        "length": {
          "getterValue": 3,
          "getterPrototypeLevel": 1,
          "enumerable": true,
          "writable": true
        }
      }
    }
  }
```

The `key` property is missing (this is probably because `Storage#key`)

**Comment by MikeRatcliffe** [comment]
@nchevobbe: I have already spoken with the platform g(irl|uy)s and they say it is not a bug.

**Comment by nchevobbe** [comment]
fixed in https://github.com/devtools-html/devtools-core/pull/1046
